File: flutter/challenge/lib/model/couponModel.dart
class CouponModel {
  String id;
	String description;
	String seller;
	String image;
	String expiresAt; 

  CouponModel({
    this.id,
    this.description,
    this.seller,
    this.image,
    this.expiresAt
  });

  factory CouponModel.fromJson(Map<String, dynamic> json) {
    return CouponModel(
        id: json["id"],
        description: json["description"],
        seller: json["seller"],
        image: json["image"],
        expiresAt: json["expiresAt"]);
  }
}

File: src/main/java/cl/talavera/api/data/model/CouponModel.java
package cl.talavera.api.data.model;

import cl.talavera.api.core.domain.Coupon;
import lombok.Data;

import java.text.DateFormat;
import java.util.Date;

@Data
public class CouponModel {
    private String id;
    private Date expiresAt;
    private String seller;
    private String description;
    private String image;

    public static Coupon CouponFrom(CouponModel c, DateFormat f) {
        return Coupon.builder()
                .id(c.getId())
                .expires(f.format(c.getExpiresAt()))
                .seller(c.getSeller())
                .image(c.getImage())
                .description(c.getDescription())
                .build();
    }
}
